gtk4/src/expression.rs

```diff
@@ -182,6 +182,56 @@ impl glib::value::ToValueOptional for Expression {
     }
 }
 
+/// Trait containing convenience methods in creating [`gtk::PropertyExpression`] that
+/// looks up a property of a [`glib::Object`].
+pub trait PropExpr {
+    /// Create an expression looking up an object's property
+    fn property_expression(&self, property_name: &str) -> Expression;
+
+    /// Create an expression looking up an object's property with a weak reference.
+    fn weak_property_expression(&self, property_name: &str) -> Expression;
+}
+
+impl<T: IsA<glib::Object>> PropExpr for T {
+    fn property_expression(&self, property_name: &str) -> Expression {
+        let obj_expr = crate::ConstantExpression::new(self).upcast();
+        crate::PropertyExpression::new(T::static_type(), Some(&obj_expr), property_name).upcast()
+    }
+
+    fn weak_property_expression(&self, property_name: &str) -> Expression {
+        let obj_expr = crate::ObjectExpression::new(self).upcast();
+        crate::PropertyExpression::new(T::static_type(), Some(&obj_expr), property_name).upcast()
+    }
+}
+
+/// Trait containing convenience methods to create chains of [`gtk::PropertyExpression`]
+/// or [`gtk::ClosureExpression`].
+pub trait ChainExpr {
+    /// Create a [`gtk::PropertyExpression`] that looks up for `property_name`
+    /// with self as parameter
+    fn property_expression(&self, property_name: &str) -> Expression;
```

Comment:
```suggestion
    fn property_expression(&self, property_name: &str) -> PropertyExpression;
```

Maybe? And `ClosureExpression` for the other function

Reply:
also with the `PropExpr/GObjectPropertyExpressionExt` i think?

Reply:
How do you mean?

Reply:
the `weak_property_expression` and the `property_expression` of the other trait should also return PropertyExpression?

Reply:
Oh, yes

Reply:
Done :D